test_evm_eth_compliance/src/statetest/runner.rs

```diff
@@ -0,0 +1,377 @@
+                        || pre_test.pre_owners.contains(&owner_address_str)
+                } else {
+                    false
+                }
+            },
+        )
+    });
+
+    if rval {
+        warn!(
+            "Skipping Pre Test test_name: '{:#?}', owner_address: '{:#?}'",
+            test_name, owner_address,
+        );
+    }
+
+    rval
+}
+
+fn skip_post_test(test_name: &str, chain_spec: &SpecName, test_id: usize) -> bool {
+    let rval = SKIP_TESTS.state.iter().any(|state_test| {
+        state_test.post_tests.as_ref().map_or_else(
+            || false,
+            |maybe_post_tests| {
+                if let Some(post_test) = maybe_post_tests.get(test_name) {
+                    chain_spec == &post_test.chain_spec
+                        && (post_test.sub_numbers[0] == "*"
+                            || post_test.sub_numbers.contains(&test_id.to_string()))
+                } else {
+                    false
+                }
+            },
+        )
+    });
+
+    if rval {
+        warn!(
+            "Skipping Post Test test_name: '{:#?}', chain_spec: '{:#?}', test_id: {:#?}",
+            test_name, chain_spec, test_id
+        );
+    }
+
+    rval
+}
+
+fn execute_test_suit(path: &Path, elapsed: &Arc<Mutex<Duration>>) -> Result<(), TestError> {
+    let json_reader = std::fs::read(path).unwrap();
+    let suit: TestSuit = serde_json::from_reader(&*json_reader)?;
+
+    let store = MemoryBlockstore::new();
+    let test_vm = VM::new_with_singletons(&store);
+    let account = create_accounts(&test_vm, 1, TokenAmount::from_whole(10_000))[0];
+    let mut pre_contract_cache: HashMap<B160, Address> = HashMap::new();
+
+    let timer = Instant::now();
+
+    for (name, unit) in suit.0.iter() {
+        // info!("{:#?}:{:#?}", name, unit);
+
+        // TODO :: Process env block, is it needed for FEVM context ?
+
+        // Process the "pre" block & deploy the contracts
+        for (_test_id, (address, info)) in unit.pre.iter().enumerate() {
+            assert!(
+                info.code.len() < MAX_CODE_SIZE,
+                "{}",
+                format!(
+                    "EVM byte code length ({}) is exceeding the maximum allowed of {MAX_CODE_SIZE}",
+                    info.code.len()
+                )
+            );
+
+            assert!(
+                info.code.first() != Some(&0xEF),
+                "EIP-3541: Contract code starting with the 0xEF byte is disallowed.",
+            );
+
+            let create_result = test_vm
+                .apply_message(
+                    account,
+                    EAM_ACTOR_ADDR,
+                    TokenAmount::zero(),
+                    fil_actor_eam::Method::CreateAccount as u64,
+                    Some(fil_actor_eam::InitAccountParams {
+                        eth_address: EthAddress(address.as_bytes().try_into().unwrap()),
+                    }),
+                )
+                .unwrap();
```

Comment:
We no longer create accounts in this manner, but rather just create an EthAccount actor at the designated address with the f4 delegated address in the init actor.